sdk/src/main/java/com/qonversion/android/sdk/internal/QonversionInternal.kt

```diff
@@ -170,6 +171,28 @@ internal class QonversionInternal(
         )
     }
 
+    override fun purchase(
+        context: Activity,
+        product: QProduct,
+        options: QPurchaseOptions,
+        callback: QonversionEntitlementsCallback
+    ) {
+        productCenterManager.purchase(context, product, options, callback)
+    }
+
+    override fun updatePurchase(
+        context: Activity,
+        product: QProduct,
+        options: QPurchaseOptions,
+        callback: QonversionEntitlementsCallback
+    ) {
+        productCenterManager.purchaseProduct(
+            context,
+            PurchaseModelInternal(product, options),
+            mainEntitlementsCallback(callback)
+        )
+    }
```

Comment:
The first method calls the proxy method, while the second calls the exact function. Let's rewrite the `purchase` so that is would also call the only `purchaseProduct` method. Due to that mess we also lost the `mainEntitlementsCallback` wrapping.

```suggestion
    override fun purchase(
        context: Activity,
        product: QProduct,
        options: QPurchaseOptions,
        callback: QonversionEntitlementsCallback
    ) {
        productCenterManager.purchaseProduct(
            context,
            PurchaseModelInternal(product, options)
            mainEntitlementsCallback(callback)
        )
    }

    override fun updatePurchase(
        context: Activity,
        product: QProduct,
        options: QPurchaseOptions,
        callback: QonversionEntitlementsCallback
    ) {
        productCenterManager.purchaseProduct(
            context,
            PurchaseModelInternal(product, options),
            mainEntitlementsCallback(callback)
        )
    }
```

Btw, why would we even need both these functions while they do exactly the same? Let's maybe leave the only one?